File: apps/frontend/src/entities/job/model/jobStatus.ts
import type { JobStatus } from './jobTypes';

export const statusLabel: Record<JobStatus, string> = {
  PENDING: 'Preparando',
  UPLOAD_REQUESTED: 'Listo para cargar',
  UPLOADED: 'Archivo cargado',
  QUEUED: 'En cola',
  PROCESSING: 'Procesando',
  COMPLETED: 'Completado',
  FAILED: 'Fallido',
  EXPIRED: 'Expirado',
  DELETED: 'Eliminado',
};

export function isActiveJobStatus(status: JobStatus): boolean {
  return ['PENDING', 'UPLOAD_REQUESTED', 'UPLOADED', 'QUEUED', 'PROCESSING'].includes(status);
}

File: apps/api/src/morphix_api/domain/value_objects/job_status.py
from __future__ import annotations

from enum import StrEnum


class JobStatus(StrEnum):
    pending = "PENDING"
    upload_requested = "UPLOAD_REQUESTED"
    uploaded = "UPLOADED"
    queued = "QUEUED"
    processing = "PROCESSING"
    completed = "COMPLETED"
    failed = "FAILED"
    expired = "EXPIRED"
    deleted = "DELETED"

    @property
    def is_active(self) -> bool:
        return self in {
            JobStatus.pending,
            JobStatus.upload_requested,
            JobStatus.uploaded,
            JobStatus.queued,
            JobStatus.processing,
        }
